sail_runtime/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+execute_process(
+    COMMAND ${SAIL_BIN} --dir
+    OUTPUT_VARIABLE sail_dir
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    COMMAND_ERROR_IS_FATAL ANY
+)
+
+add_library(sail_runtime
+    "${sail_dir}/lib/elf.c"
+    "${sail_dir}/lib/elf.h"
+    "${sail_dir}/lib/rts.c"
+    "${sail_dir}/lib/rts.h"
+    "${sail_dir}/lib/sail.c"
+    "${sail_dir}/lib/sail.h"
+    "${sail_dir}/lib/sail_failure.c"
+    "${sail_dir}/lib/sail_failure.h"
+    "${sail_dir}/lib/sail_coverage.h"
+    "${sail_dir}/lib/sail_state.h"
+)
+
+target_include_directories(sail_runtime
+    PUBLIC "${sail_dir}/lib"
+)
+
+if (COVERAGE)
+    find_package(Threads REQUIRED)
+
+    # Note, you must run `make` in the ${sail_dir}/lib/coverage
```

Comment:
Probably worth improving this in the future (e.g. building it automatically if not present), but not for this PR.

Reply:
Yes, definitely! See https://github.com/rems-project/sail/issues/893